### PR TITLE
fix(misconf): escape all special sequences

### PR DESCRIPTION
**Pull request by nikpivkin**
## Related issues
- Close https://github.com/aquasecurity/trivy/issues/7557

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/iac/terraform/resource_block.go [hunk 1]
```diff
@@ -171,3 +165,29 @@ func renderMap(val map[string]any) string {
...
+				continue
+			}
+
+			sb.WriteRune(r)
+		} else {
+			sb.WriteRune(r)
+		}
+	}
+	return sb.String()
+}
```
Comment (simar7) [inline pkg/iac/terraform/resource_block.go:193]: I wonder if hashicorp already has this logic somewhere that we could borrow?

Comment (nikpivkin) [reply pkg/iac/terraform/resource_block.go:193]: internal only

Comment (simar7) [reply pkg/iac/terraform/resource_block.go:193]: ah bummer

Comment (simar7) [reply pkg/iac/terraform/resource_block.go:193]: could you cite the source in that case?

Comment (nikpivkin) [reply pkg/iac/terraform/resource_block.go:193]: There's not much left of the source https://github.com/hashicorp/hcl/blob/main/hclwrite/generate.go#L367-L373

**Review by simar7 — APPROVED** [review]

**Pull request merged**